src/ocl/convolutionocl.cpp

```diff
@@ -3672,24 +3610,27 @@ std::size_t ConvolutionDescriptor::GetWrwSolutionWorkspaceSize(Handle& handle,
     MIOPEN_LOG_I2("solver_id = " << solver_id.ToString());
     if(!solver_id.IsValid())
         MIOPEN_THROW(miopenStatusBadParm, "invalid solution id = " + solver_id.ToString());
-    if(solver_id != solver::Id::gemm() && solver_id != solver::Id::fft())
-    {
-        auto sol = solver_id.GetSolver();
-        auto problem =
-            ProblemDescription{xDesc, dwDesc, dyDesc, *this, conv::Direction::BackwardWeights};
-        auto ctx = ConvolutionContext{problem};
-        ctx.SetStream(&handle);
-        ctx.DetectRocm();
-        if(sol.IsApplicable(ctx))
-            return sol.GetWorkspaceSize(ctx);
-        else
-        {
-            MIOPEN_THROW(miopenStatusBadParm,
-                         "The supplied solution id: " + solver_id.ToString() +
-                             " is not applicable to the current problem");
-        }
-    }
-    return GetWrwSolutionWorkspaceSizeFallback(handle, dyDesc, xDesc, dwDesc, solver_id);
+
+    if(solver_id == solver::Id::fft())
+        MIOPEN_THROW(miopenStatusBadParm,
+                     "The supplied solution id: " + solver_id.ToString() +
+                         " is not applicable to the current problem");
```

Comment:
How did we reach to a conclusion about problem applicability here?

Reply:
FFT is not applicable for WrW.